src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -180,7 +180,7 @@ private bool IsSimpleImplicitExpression(LSPDocumentSnapshot documentSnapshot, Co
             var navigator = _textStructureNavigator.GetTextStructureNavigator(snapshot.TextBuffer);
             var line = snapshot.GetLineFromLineNumber(request.Position.Line);
             var absoluteIndex = line.Start + request.Position.Character;
-            if (absoluteIndex >= snapshot.Length)
+            if (absoluteIndex > snapshot.Length)
```

Comment:
Thank god the word navigator respects the end of document location as a valid location!